components/formats-gpl/src/loci/formats/in/OlympusINIReader.java

```diff
@@ -0,0 +1,950 @@
+    baseHeight = iniConf.get(0).totalHeight;
+    LOGGER.debug("Total number of iniConf: " + iniConf.size());
+    core.clear();
+    for (int i = 0; i < 4; i++) {
+      if (iniConf.get(i).found) {
+       core.add(new CoreMetadata());
+      }
+    }
+    int totalCores = 0;
+    for (int i = 0; i < 4; i++) {
+      if (iniConf.get(i).found) {
+        CoreMetadata m = core.get(totalCores);
+        m.sizeX = (int) iniConf.get(i).totalWidth;
+        m.sizeY = (int) iniConf.get(i).totalHeight;
+        m.sizeZ = 1;
+        m.sizeC = 3;
+        m.sizeT = 1;
+        m.rgb = true;
+        m.imageCount = 1;
+        m.dimensionOrder = "XYCZT";
+        m.pixelType = FormatTools.UINT8;
+        m.interleaved = true;
+        m.falseColor = false;
+        m.thumbnail = false;
+        m.orderCertain = true;
+        m.bitsPerPixel = 8;
+        m.thumbSizeX = 0;
+        m.thumbSizeY = 0;
+        for (int j = i+1; j < 4; j++) {
+          if (iniConf.get(j).found) {
+            m.thumbSizeX = (int) iniConf.get(j).totalWidth;
+            m.thumbSizeY = (int) iniConf.get(j).totalHeight;
+          }
+        }
+        m.littleEndian = false;
+        m.indexed = false;
+        if (i==0 && totalLevels > 2) 
+        {
+          //m.resolutionCount = 1;
+          m.resolutionCount = totalLevels;
+        }
+        else 
+        {
+          m.thumbnail = true;
+          m.resolutionCount = 1;
+        }
+        totalCores++;
+      }
+    }
+    MetadataStore store = makeFilterMetadata();
+    MetadataTools.populatePixels(store, this);
+
+    String path = new Location(currentId).getAbsoluteFile().getName();
+
+    if (totalLevels > 3)
+    {
+      DecimalFormat df = new DecimalFormat("##.##");
+      df.setMaximumFractionDigits(2);
+      //store.setImageName(path+" "+df.format(magnification) + "x", 0);
+      store.setImageName("1 ("+df.format(magnification)+"x)", 0);
+    }
+    if (totalLevels > 2)
+    {
+      //store.setImageName(path+" 20x", totalLevels-3);
+      store.setImageName("2 (20x)", totalLevels-3);
+    }
+    if (totalLevels > 1)
+    {
+      //store.setImageName(path+" 5x", totalLevels-2);
+      store.setImageName("3 (5x)", totalLevels-2);
+    }
+    if (totalLevels > 0)
+    {
+      //store.setImageName(path+" 1.25x", totalLevels-1);
+      store.setImageName("4 (1.25x)", totalLevels-1);
+    }
+
+    //CoreMetadata ms0 = core.get(0);
+
+    if (getMetadataOptions().getMetadataLevel() != MetadataLevel.MINIMUM) 
+    {
+      String instrumentID = MetadataTools.createLSID("Instrument", 0);
+      store.setInstrumentID(instrumentID, 0);
+
+      String objectiveID = MetadataTools.createLSID("Objective", 0, 0);
+      store.setObjectiveID(objectiveID, 0, 0);
+      store.setObjectiveNominalMagnification(magnification, 0, 0);
+
+      store.setExperimenterID(MetadataTools.createLSID("Experimenter", 0), 0);
+
+
+      totalCores = 0;
+      for (int fileNum=0; fileNum < 4; fileNum++) 
+      {
```

Comment:
So in this case we only want to populate Pixel metadata based on the expected length of the series. So you could use:
`for (int fileNum=0; fileNum < getSeriesCount(); fileNum++) `